Atividades/Em casa/alexandra-ribeiro/clinicaVet.js

```diff
@@ -0,0 +1,204 @@
+        }
+    }
+
+    brincar() {
+        super.brincar()
+    }
+
+    acariciar() { 
+        if(this.social) {
+            console.log("ron ron ron") 
+        } else {
+            console.log("silvando")
+        }
+    }
+
+    alimentar(comida) {
+        const comidasFavoritas = ["frango", "sache", "peixe"];
+        if(comidasFavoritas.includes(comida)) {
+            console.log(`${this.nome} está sendo alimentado com ${comida}`);
+            this.social = true
+        }
+    }
+}
+
+let nida = new Gato("Nida", 1.5, "preto", true, true, true);
+let margot = new Gato("Margot", 3, "branca", true, true, false);
+
+class Cachorro extends Animal {
+    #ferido
+    constructor(nome, idade, cor, castrado, raca, ferido) {
+        super(nome, idade, cor)
+        this.castrado = castrado;
+        this.raca = raca;
+        this.#ferido = ferido
+    }
+
+    latir() { console.log("au au au")}
+
+    vacinar(vacina) {
+        this.historico.vacinas.push(vacina)
+        this.consultar()
+    }
+
+    consultar() {
+        super.consultar();
+    }
+    
+    brincar() {
+        if(this.#ferido) {
+            console.log(`${this.nome} está ferido e não quer brincar`)
+        } else {
+            console.log(`${this.nome} está brincando com a bolinha`)
+        }
+    }
+
+    estaFerido() { return this.#ferido }
+
+    castrar() {
+        if(this.castrado) {
+            console.log("O cachorro já foi castrado")
+        } else{
+            this.castrado = true;
+            console.log("O cachorro foi castrado")
+            this.consultar();
+        }
+    }
+
+    acariciar() {
+        super.acariciar();
+    }
+
+    alimentar() {
+        super.alimentar();
+    }
+}
+
+let sushi = new Cachorro("Sushi", 48, "bege", true, "Pug", true);
+let saroo = new Cachorro("Saroo", 60, "marrom", true, "Yorkshire", false);
+
+class Hamster extends Animal {
+    constructor(nome, idade, cor, tipo) {
+        super(nome, idade, cor)
+        this.tipo = tipo
+        this.consultas = []
+    }
+
+    consultar() {
+        super.brincar();
+        console.log(`${this.nome} está mais calmo e pode fazer a consulta`)
+        super.consultar();
+    }
+
+    brincar() {
+        console.log(`${this.nome} está brincando na rodinha`)
+    }
+
+    acariciar() {
+        super.acariciar();
+    }
+
+    alimentar() {
+        super.alimentar();
+    }
+}
+
+let cookie = new Hamster("Cookie", 3, "marrom e branco", "Sírio")
+
+class Papagaio extends Animal {
+    constructor(nome, idade, cor) {
+        super(nome, idade, cor)
+        this.consultas = [];
+    }
+
+    falar() { console.log("Olá")}
+
+    consultar() {
+        super.consultar();
+        this.brincar();
+    }
+
+    brincar() {
+        return this.falar()
+    }
+
+    acariciar() {
+        return this.falar()
+    }
+
+    alimentar() {
+        super.alimentar()
+    }
+}
+
+let loro = new Papagaio("Loro", 2, "verde")
+
+function chamar(animais) {
```

Comment:
Adorei sua lógica do for e o uso do if,  seu código ficou mega organizado dessa forma